c7n/resources/s3.py

```diff
@@ -1052,13 +1053,49 @@ def process_resource_set(self, resource_set, tags):
                     Bucket=r['Name'], Tagging={'TagSet': tag_set})
 
 
+@actions.register('mark-for-op')
+class MarkBucketForOp(TagDelayedAction):
+
+    schema = type_schema(
+        'mark-for-op', rinherit=TagDelayedAction.schema)
+
+    def process_resource_set(self, resource_set, tags):
+        client = local_session(self.manager.session_factory).client('s3')
```

Comment:
you could also pull out the common implementation to a helper method to invoke from your subclass. if you go the mixin class route, you need to play with subclass inheritance ordering to get the proper behavior.